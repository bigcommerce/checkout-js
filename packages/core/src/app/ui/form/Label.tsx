import classNames from 'classnames';
import React, { FunctionComponent, LabelHTMLAttributes } from 'react';

export interface LabelProps extends LabelHTMLAttributes<HTMLLabelElement> {
    testId?: string;
    hidden?: boolean;
    useFloatingLabel?: boolean;
}

const Label: FunctionComponent<LabelProps> = ({
    children,
    className,
    hidden,
    testId,
    useFloatingLabel,
    ...rest
}) => (
    <label
        { ...rest }
        className={ classNames(
            { 'floating-label': useFloatingLabel },
            className || 'form-label',
            { 'is-srOnly': hidden },
            'optimizedCheckout-form-label'
        ) }
        data-test={ testId }
    >
        { children }
    </label>
);

export default Label;
