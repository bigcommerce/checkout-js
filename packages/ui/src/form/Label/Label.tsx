import classNames from 'classnames';
import React, { FunctionComponent, LabelHTMLAttributes } from 'react';

export interface LabelProps extends LabelHTMLAttributes<HTMLLabelElement> {
    testId?: string;
    hidden?: boolean;
}

const Label: FunctionComponent<LabelProps> = ({ children, className, hidden, testId, ...rest }) => (
    <label
        {...rest}
        className={classNames(
            className || 'form-label',
            { 'is-srOnly': hidden },
            'optimizedCheckout-form-label',
        )}
        data-test={testId}
    >
        {children}
    </label>
);

export default Label;
