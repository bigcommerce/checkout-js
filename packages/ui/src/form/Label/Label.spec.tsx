import { shallow } from 'enzyme';
import React from 'react';

import Label from './Label';

describe('Label', () => {
    it('matches snapshot', () => {
        expect(shallow(<Label>Hello world</Label>)).toMatchSnapshot();
    });

    it('renders component with text', () => {
        expect(shallow(<Label>Hello world</Label>).text()).toBe('Hello world');
    });

    it('renders component with test ID', () => {
        expect(shallow(<Label testId="test">Hello world</Label>).prop('data-test')).toBe('test');
    });
});
