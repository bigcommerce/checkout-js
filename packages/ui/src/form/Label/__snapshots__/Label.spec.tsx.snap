// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label matches snapshot 1`] = `
<label
  className="form-label optimizedCheckout-form-label"
>
  Hello world
</label>
`;
